connect/transforms/src/test/java/org/apache/kafka/connect/transforms/field/SingleFieldPathTest.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    @Test void shouldFindValueInMap() {
+        Map<String, Object> foo = new HashMap<>();
+        foo.put("bar", 42);
+        foo.put("baz", null);
+        Map<String, Object> map = new HashMap<>();
+        map.put("foo", foo);
+
+        assertEquals(42, pathV2("foo.bar").valueFrom(map));
+        assertNull(pathV2("foo.baz").valueFrom(map));
+    }
+
+    @Test void shouldReturnNullValueWhenFieldNotFoundInMap() {
+        Map<String, Object> foo = new HashMap<>();
+        foo.put("bar", 42);
+        foo.put("baz", null);
+        Map<String, Object> map = new HashMap<>();
+        map.put("foo", foo);
+
+        assertNull(new SingleFieldPath("un.known", FieldSyntaxVersion.V2).valueFrom(map));
+        assertNull(new SingleFieldPath("foo.unknown", FieldSyntaxVersion.V2).valueFrom(map));
+        assertNull(new SingleFieldPath("unknown", FieldSyntaxVersion.V2).valueFrom(map));
+        assertNull(new SingleFieldPath("foo.baz", FieldSyntaxVersion.V2).valueFrom(map));
+        assertNull(new SingleFieldPath("foo.baz.inner", FieldSyntaxVersion.V2).valueFrom(map));
+    }
+
+    @Test void shouldFindValueInStruct() {
+        SchemaBuilder bazSchema = SchemaBuilder.struct()
+            .field("inner", Schema.STRING_SCHEMA);
+        SchemaBuilder barSchema = SchemaBuilder.struct()
+            .field("bar", Schema.INT32_SCHEMA)
+            .field("baz", bazSchema.optional());
+        Schema schema = SchemaBuilder.struct().field("foo", barSchema).build();
+        Struct foo = new Struct(barSchema)
+            .put("bar", 42)
+            .put("baz", null);
+        Struct struct = new Struct(schema).put("foo", foo);
+
+        assertEquals(42, pathV2("foo.bar").valueFrom(struct));
+        assertNull(pathV2("foo.baz").valueFrom(struct));
+    }
+
+    @Test void shouldReturnNullValueWhenFieldNotFoundInStruct() {
+        SchemaBuilder bazSchema = SchemaBuilder.struct()
+            .field("inner", Schema.STRING_SCHEMA);
+        SchemaBuilder barSchema = SchemaBuilder.struct()
+            .field("bar", Schema.INT32_SCHEMA)
+            .field("baz", bazSchema.optional());
+        Schema schema = SchemaBuilder.struct().field("foo", barSchema).build();
+        Struct foo = new Struct(barSchema)
+            .put("bar", 42)
+            .put("baz", null);
+        Struct struct = new Struct(schema).put("foo", foo);
+
+        assertNull(new SingleFieldPath("un.known", FieldSyntaxVersion.V2).valueFrom(struct));
+        assertNull(new SingleFieldPath("foo.unknown", FieldSyntaxVersion.V2).valueFrom(struct));
+        assertNull(new SingleFieldPath("unknown", FieldSyntaxVersion.V2).valueFrom(struct));
+        assertNull(new SingleFieldPath("foo.baz", FieldSyntaxVersion.V2).valueFrom(struct));
+        assertNull(new SingleFieldPath("foo.baz.inner", FieldSyntaxVersion.V2).valueFrom(struct));
+    }
+
+    private static SingleFieldPath pathV2(String path) {
+        return new SingleFieldPath(path, FieldSyntaxVersion.V2);
+    }
```

Comment:
We can simplify things even further by piggybacking off of this with a few extra utility methods:

```java
    private static Field pathV2Field(String path, Schema schema) {
        return pathV2(path).fieldFrom(schema);
    }

    private static Object pathV2Value(String path, Struct struct) {
        return pathV2(path).valueFrom(struct);
    }

    private static Object pathV2Value(String path, Map<String, Object> map) {
        return pathV2(path).valueFrom(map);
    }
```

I think we can replace all direct invocations of `pathV2` with these wrapper methods.

Reply:
Not sure if the proposed methods add much. I'd argue the `pathV2()` util is enough to make the invocations more readable.

Reply:
I found it easier to read/review after doing the refactoring, but I don't consider this a blocker. Can always see how things pan out with future PRs when we add more functionality to the `SingleFieldPath` class.